### DISCUSSION TITLE
DateField that starts with empty date and updated programmatically

### DISCUSSION THREAD

**Discussion by brandonpittman** [adobe/react-spectrum#4303] [Q&A] [answered]
As far as I can see, `useDateField` takes a `CalendarDate | undefined` for its value. But if it starts as undefined and it later updated, you get the old React warning about controlled/uncontrolled changes. Since a `CalendateDate` can't be empty, how would you have a date field that starts empty and can then be updated by the parent?

**Comment by devongovett** [comment, marked as answer]
The value can be `null` to set it to an empty value. However, in the current release TypeScript strict mode complains about this so you may need to `@ts-ignore`. This is already fixed in the latest nightly, and will be part of our upcoming release very soon.

**Reply by brandonpittman** [reply]
Thanks. `null` was okay, despite TypeScript's complaints.
